templates/ingress.yaml

```diff
@@ -0,0 +1,34 @@
+{{- if .Values.ingress.enabled }}
+{{- range .Values.ingress.hosts }}
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: {{ template "storageos.fullname" $ }}
+  namespace: {{ $.Values.namespace }}
+  labels:
+    app: {{ template "storageos.name" $ }}
+    chart: {{ template "storageos.chart" $ }}
+    release: {{ $.Release.Name }}
+    heritage: {{ $.Release.Service }}
+  annotations:
+    {{- if .tls }}
+    ingress.kubernetes.io/secure-backends: "true"
+    {{- end }}
+    {{- range $key, $value := .annotations }}
+    {{ $key }}: {{ $value | quote }}
+    {{- end }}
+spec:
+  rules:
+  - host: {{ .name }}
+  backend:
```

Comment:
I have always added an http section after host. Such as the following 
```
spec:
  rules:
    - host: test.mydomain.com
      http:
        paths:
         - path: /
           backend:
             serviceName: svcname
             servicePort: 80
```
I'm deploying an nginx controller to double check that we make this work in different envs.
Nginx ingress-controller examples also shows this format. https://github.com/kubernetes/ingress-nginx/blob/master/docs/examples/tls-termination/ingress.yaml

Reply:
That should work as well. This `backend` exists here so that by default, if no host config is defined, we use our service.